src/containers/BooksList.js

```diff
@@ -3,39 +3,52 @@ import React from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import Book from '../components/Book';
-import { removeBook } from '../actions';
+import { removeBook, changeFilter } from '../actions';
+import CategoryFilter from '../components/CategoryFilter';
 
 const BooksList = (props) => {
-  const { books } = props;
+  const { books, fil } = props;
 
   const handleRemoveBook = (book) => {
     props.dispatch(removeBook(book));
   };
 
+  const handleFilterChange = (e) => {
+    props.dispatch(changeFilter(e.target.value));
+  };
```

Comment:
- [ ] Kindly use redux to dispatch the `removeBook `and `changeFilter ` by using the useDispatch hook or mapDispatchToProps. Handling it in this manner is not advised, encapsulating the dispatch logic into a function makes the implementation more declarative. Dispatching an action and letting the Redux store handle the data flow is how to conventionally implement the behavior.